### ISSUE TITLE
Handle `not` in `:condition-always-true` / new linter - `:condition-always-false`

### ISSUE THREAD

**Issue by skynet-gh** [clj-kondo/clj-kondo#2373]
[ Leave this text in if you want to promote it. ]

To upvote this issue, give it a thumbs up. See [this list](https://github.com/clj-kondo/clj-kondo/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc) for the most upvoted issues.

- [x] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

The [`:condition-always-true` linter](https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#condition-always-true) is very helpful, but it "breaks" when inverting the check with `not` for example.

**Describe the solution you'd like**

It'd be great if `:condition-always-true` could work in cases like `(when (not true) ...)` as well, or maybe a separate linter `:condition-always-false`.

**Describe alternatives you've considered**

Sometimes the boolean logic is just setup in a way where you get `(not ...)` and so the linter fails to catch errors in there.

**Additional context**

`repro.clj`:

```clj
(ns repro)

(defn x? [& _]
  "z")

(when (not x?)
  (println "y"))
```

```sh
$ clj-kondo --lint repro.clj
linting took 5ms, errors: 0, warnings: 0
```

**Comment by borkdude** [comment]
Hmm yes, perhaps we should rename `:condition-always-true` to `:condition-...` what actually? `:condition-always-constant`? but `constant` already implies always :) If you can recommend a better name, please do.

**Comment by borkdude** [comment]
Btw, one thing that prevented me to handle the false case in a Slack discussion was the following:

``` clojure
(def debug false)

(if debug ...)
```

It would be annoying if clj-kondo complained about that particular usage.

**Comment by borkdude** [comment]
Btw, this does warn:

```
(when-not x?
  (println "y"))
```

**Comment by skynet-gh** [comment]
@borkdude maybe `:condition-is-constant`? Or maybe `:boolean-test-is-constant`? (since the Clojure docs for `if` and `when` call it `test`)

For your example of a debug flag, maybe it'd be possible to add config to the linter with exception names? Like `:exclude [debug]`

**Comment by borkdude** [comment]
Ah wait, this already doesn't warn:

```
(def DEBUG true)

(if DEBUG 1 2)
```

but `(if #'DEBUG 1 2)` does warn. Let me check how that was implemented. We can do the same for `:condition-always-false`

**Comment by borkdude** [comment]
OK, the current way `:condition-always-true` is pretty limited, it will only warn in this these cases:

```
(if #'inc .. ..) ;; var
```

and

```
(if inc 1 2)
```

so only on passing a fn and a var, which are commonly reported mistakes. I guess `(not inc)` was a variation on that common theme?

**Comment by borkdude** [comment]
condition-always-true has been widened to more stuff recently. I guess we could use the same linter in `not`:

``` clojure
(let [x? ""]
  (when (not x?)
    (println "y")))
```
would just warn about `x?` being always true in not?

cc @NoahTheDuke
